src/main/extension/extension_install.cpp

```diff
@@ -191,10 +197,22 @@ void ExtensionHelper::InstallExtensionInternal(DBConfig &config, ClientConfig *c
 	throw BinderException("Remote extension installation is disabled through configuration");
 #else
 
-	string default_endpoint = "http://extensions.duckdb.org";
 	string versioned_path = "/${REVISION}/${PLATFORM}/${NAME}.duckdb_extension.gz";
 	string custom_endpoint = client_config ? client_config->custom_extension_repo : string();
-	string &endpoint = !custom_endpoint.empty() ? custom_endpoint : default_endpoint;
+	string endpoint;
+	if (!repository.empty()) {
+		endpoint = repository;
+	} else if (!custom_endpoint.empty()) {
+		endpoint = custom_endpoint;
+	} else {
+		endpoint = "http://extensions.duckdb.org";
+	}
+
+	if (StringUtil::Contains(endpoint, "https://")) {
+		// Downgrade https to http
+		endpoint = "http://" + endpoint.substr(8, endpoint.size());
+	}
```

Comment:
Unsure of this, I think it might be helpful, but also confusing.
Either removing entirely or adding a better comment than mine